rocketpy/mathutils/function.py

```diff
@@ -2817,6 +2817,76 @@ def compose(self, func, extrapolate=False):
                 extrapolation=self.__extrapolation__,
             )
 
+    def savetxt(
+        self,
+        filename,
+        lower=None,
+        upper=None,
+        samples=None,
+        fmt="%.6f",
+        delimiter=",",
+        newline="\n",
+        encoding=None,
+    ):
+        """Save a Function object to a text file. The first line is the header
+        with inputs and outputs. The following lines are the data. The text file
+        can have any extension, but it is recommended to use .csv or .txt.
+
+        Parameters
+        ----------
+        filename : str
+            The name of the file to be saved, with the extension.
+        lower : float or int, optional
+            The lower bound of the range for which data is to be generated.
+            This is required if the source is a callable function.
+        upper : float or int, optional
+            The upper bound of the range for which data is to be generated.
+            This is required if the source is a callable function.
+        samples : int, optional
+            The number of sample points to generate within the specified range.
+            This is required if the source is a callable function.
+        fmt : str, optional
+            The format string for each line of the file, by default "%.6f".
+        delimiter : str, optional
+            The string used to separate values, by default ",".
+        newline : str, optional
+            The string used to separate lines in the file, by default "\n".
+        encoding : str, optional
+            The encoding to be used for the file, by default None (which means
+            using the system default encoding).
+
+        Raises
+        ------
+        ValueError
+            Raised if `lower`, `upper`, and `samples` are not provided when
+            the source is a callable function. These parameters are necessary
+            to generate the data points for saving.
+        """
+        # create the header
+        header_line = delimiter.join(self.__inputs__ + self.__outputs__)
+
+        # create the datapoints
+        if callable(self.source):
+            if lower is None or upper is None or samples is None:
+                raise ValueError(
+                    "If the source is a callable, lower, upper and samples"
+                    + " must be provided."
+                )
+            # Generate the data points using the callable
+            x = np.linspace(lower, upper, samples)
+            data_points = np.column_stack((x, self.source(x)))
+        else:
+            # If the source is already an array, use it as is
+            data_points = self.source
+
+            if lower and upper and samples:
+                data_points = self.set_discrete(lower, upper, samples).source
```

Comment:
I advise deepcopying with `copy.deepcopy` and then performing discretization over this temporary object, since the `set_discrete` method is a instance mutator: it changes the `Function` source. This is unlikely to be the intention or expected when running this method, I believe.

Reply:
I'm currently waiting for the #519 resolution.

Reply:
Solved here: [696e709](https://github.com/RocketPy-Team/RocketPy/pull/514/commits/696e709945fea9d957a2615ef72c5f570218260f)

The #519 made my life much easier here, thank you @phmbressan !!